chapter_13_database_layer_validation.asciidoc

```diff
@@ -241,8 +258,15 @@ been passed an error variable, and if so, we do this:
 
 ((("Bootstrap", "documentation")))
 ((("form control classes (Bootstrap)")))
-Take a look at the http://getbootstrap.com/css/#forms[Bootstrap docs] for more
+Take a look at the https://getbootstrap.com/docs/5.3/forms/validation/#server-side[Bootstrap docs] for more
 info on form controls.
+// CSANAD: these are the new docs for Bootstrap, but for some reason they begin
+//         with saying "We recommend client-side validation" which is bad.
+// Client side validation is fine for faster UI aesthetics, sure. But they are
+// not emphasizing that it is only for faster feedback and for security,
+// whatever reaches the server should be validated on the server side before
+// accepting it as an input for literally anything... Maybe we should mention
```

Comment:
maybe just a simple footnote "... and ignore their advice to prefer client-side validation.  (a) both is better and (b) they're going to say that because they're frontend people and (c) honestly if you cant do both then server-side validation is probably the one you really can't do without"